### PR TITLE
Db week3

### PR DESCRIPTION
**Pull request by Sreelatha-Vallabhaneni**
please review my home work

### CODE REVIEW

File: databases/week3/createTables.sql [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+USE MealSharing;
+
+CREATE  TABLE meal (
+id INT(10) UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+title VARCHAR(255) NOT NULL,
+description TEXT NOT NULL,
+location VARCHAR(255) NOT NULL,
+`when` DATETIME NOT NULL,
+max_reservations INT(10),
+price DECIMAL NOT NULL,
```
Comment (flemeur) [inline databases/week3/createTables.sql]: Nice that you are using a decimal here. A thing to note is that decimal defaults to `DECIMAL(10, 0)`, which will not allow for any digits after the point. I would recommend using `DECIMAL(13, 4)` instead for anything with money. See: https://rietta.com/blog/best-data-types-for-currencymoney-in/ for details

File: databases/week3/createTables.sql [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+INSERT INTO Meal(title, description, location, `When`, max_reservations, price, created_date)
+VALUES ('Pizza', 'Pizza with cheese', 'Odense', '01.09.2019', 3, '5.55', now());
+INSERT INTO Meal(title,description, location, `When`, max_reservations, price, created_date) 
+VALUES ('Soup', 'Tomato soup', 'Odense', '30.08.2019', 2, '2', '2.10.2019');
+INSERT INTO Meal(title, description, location, `When`, max_reservations, price, created_date) 
+VALUES('pizza', 'chicken pizza', 'taastrup', '11.8.2019', 4, '60', '27.11.2019');
+INSERT INTO Meal(title,description, location, `When`, max_reservations, price, created_date) 
+VALUES ('Soup', 'Tomato soup', 'Odense', '30.08.2019', 42, '2', '2.10.2029');
+INSERT INTO Meal(title,description, location, `When`, max_reservations, price, created_date) 
+VALUES ('Dhum biryani and Rød grød med fløde', 'Tomato soup', 'høetaastrup', '2029-08-30', 59, '2', '2019-10-2');
```
Comment (flemeur) [inline databases/week3/createTables.sql:45]: This gives my some errors. Firstly you need to spell the table name with lowercase in the insert statements, when the table has been defined with lowercase. Also the formats for the dates in `when` and `created_at` are not correct. It needs to be `YYYY-MM-DD HH:mm:SS` like described here https://dev.mysql.com/doc/refman/8.0/en/date-and-time-literals.html

File: databases/week3/createTables.sql [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+-- Delete a meal with any id, fx 1
+DELETE FROM Meal WHERE id = 3;
+
+
+-- Reservation
+INSERT INTO reservation (number_of_guests, meal_id, created_date) VALUES (4, 1, "2019-11-17");
+INSERT INTO reservation (number_of_guests, meal_id, created_date) VALUES (3, 4, "2019-10-10");
+INSERT INTO reservation (number_of_guests, meal_id, created_date) VALUES (5, 3, "2019-07-30");
+INSERT INTO reservation (number_of_guests, meal_id, created_date) VALUES (2, 2, "2019-9-28");
```
Comment (flemeur) [inline databases/week3/createTables.sql]: Remember that double quotes are not the standard way of specifying strings in SQL. It's better to use single quotes

File: databases/week3/createTables.sql [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+
+-- Additional queries
+
+-- Get meals that has a price smaller than a specific price fx 90
+SELECT * FROM Meal WHERE Price < 4;
+
+-- Get meals that still has available reservations
+SELECT meal.title, meal.max_reservations, reservation.number_of_guests FROM meal JOIN reservation ON meal.id = reservation.meal_id
+WHERE meal.max_reservations >  reservation.number_of_guests;
```
Comment (flemeur) [inline databases/week3/createTables.sql:122]: Great!

File: databases/week3/createTables.sql [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+-- Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+SELECT * FROM meal WHERE title LIKE '%Rød grød med fløde%'; 
+
+-- Get meals that has been created between two dates
+SELECT * FROM meal
+WHERE created_date BETWEEN '2019-01-5' AND '2030-10-20';
```
Comment (flemeur) [inline databases/week3/createTables.sql]: The exercise does not specify if the date range should be inclusive or exclusive. Using between makes the where clause inclusive. If it should be exclusive then you can also do `WHERE created_date < '2019-01-5' AND created_date > '2030-10-20'`

File: databases/week3/createTables.sql [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+-- Get reservations for a specific meal sorted by created_date
+SELECT meal.title, reservation.created_date, reservation.number_of_guests
+FROM meal JOIN reservation ON meal.id = reservation.meal_id
+WHERE meal.title LIKE '%soup%'
+ORDER BY reservation.created_date;
+
+-- Sort all meals by average number of stars in the reviews
+SELECT meal.title, AVG(review.stars) AS stars FROM meal JOIN review ON meal.id = review.meal_id 
+GROUP BY meal.title
```
Comment (flemeur) [inline databases/week3/createTables.sql]: Great!. Although be careful about using meal.title for group by. You have two different meals with a title of "pizza" which will mean that they will be group incorrectly. It's better to use meal.id for group by

**Review by flemeur — CHANGES_REQUESTED** [review]
Hi @Sreelatha-Vallabhaneni 

Nice job on the homework :+1: I've added my feedback here

File: databases/week3/createTables.sql (hunk 1 above)
Comment (Sreelatha-Vallabhaneni) [reply databases/week3/createTables.sql]: Added DECIMAL(13, 4)

File: databases/week3/createTables.sql (hunk 2 above)
Comment (Sreelatha-Vallabhaneni) [reply databases/week3/createTables.sql:45]: changed date format and table name.

File: databases/week3/createTables.sql (hunk 3 above)
Comment (Sreelatha-Vallabhaneni) [reply databases/week3/createTables.sql]: Added single quotes ' '

File: databases/week3/createTables.sql (hunk 5 above)
Comment (Sreelatha-Vallabhaneni) [reply databases/week3/createTables.sql]: changed WHERE clause to exclusive.

File: databases/week3/createTables.sql (hunk 6 above)
Comment (Sreelatha-Vallabhaneni) [reply databases/week3/createTables.sql]: changed meal.id for GROUP BY

File: databases/week3/createTables.sql (hunk 4 above)
Comment (Sreelatha-Vallabhaneni) [reply databases/week3/createTables.sql:122]: Thank you!

**Pull request merged**